src/net.cpp

```diff
@@ -671,6 +671,231 @@ void SocketSendData(CNode *pnode)
 
 static list<CNode*> vNodesDisconnected;
 
+class CNodeRef {
```

Comment:
`shared_ptr` was introduced in C++ 11, so it makes sense that the upstream PR wouldn't have included it.